api/engineapi/models.py

```diff
@@ -155,6 +155,82 @@ class DropperClaimant(Base):  # type: ignore
     )
 
 
+class RegisteredContract(Base):  # type: ignore
+    __tablename__ = "registered_contracts"
+    __table_args__ = (
+        UniqueConstraint(
```

Comment:
We need to check everywhere in actions that `contract_type` and `blockchain` pass Enum, otherwise database could have `Ethereum` or `ETHEREUM` values and UniqueConstraint will not affect.

Reply:
Right now, this is being validated at CLI level and at API level (FastAPI handler parses the type). I could mark the inputs to the API and CLI as strings and do the parsing explicitly in the actions. Would you prefer that?

Reply:
Added TODO regarding this in the code.